sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -251,30 +258,124 @@ export class AssessmentControllerService {
    * Populate the final awards in a dynamic way like disbursement schedule(estimated) awards.
    * @param disbursementReceipts disbursement receipt details.
    * @param disbursementScheduleId disbursement schedule id of the disbursement receipt(s).
-   * @param identifier identifier which is used to create dynamic data by appending grant code
+   * @param identifier identifier which is used to create dynamic data by appending award code
    * to it.
    * @returns dynamic award data of disbursement receipts of a given disbursement.
+   * @example
+   * finalAward: {
+   *    disbursementReceipt1cslf: 1000,
+   *    disbursementReceipt1csgp: 10001,
+   *    disbursementReceipt1bcsl: 1005,
+   *    disbursementReceipt1bcag: 1006,
+   *    disbursementReceipt1bgpd: 1007,
+   *    disbursementReceipt1sbsd: 1008,
+   *    disbursementReceipt2cslf: 1000,
+   *    disbursementReceipt2csgp: 10001,
+   *    disbursementReceipt2bcsl: 1005,
+   *    disbursementReceipt2bcag: 1006,
+   *    disbursementReceipt2bgpd: 1007,
+   *    disbursementReceipt2sbsd: 1008,
+      }
    */
   private populateDisbursementReceiptAwardValues(
     disbursementReceipts: DisbursementReceipt[],
-    disbursementScheduleId: number,
+    disbursementSchedule: DisbursementSchedule,
     identifier: string,
   ): Record<string, string | number> {
     const finalAward = {};
+    // Add all estimated awards to the list of receipts returned.
+    // Ensure that every estimated disbursement will be part of the summary.
+    disbursementSchedule.disbursementValues.forEach((award) => {
+      if (award.valueType === DisbursementValueType.BCTotalGrant) {
+        // BC Total grants are not part of the students grants and should not be part of the summary.
+        return;
+      }
+      const disbursementValueKey = this.createReceiptFullIdentifier(
+        identifier,
+        award.valueCode,
+      );
+      finalAward[disbursementValueKey] = null;
```

Comment:
I feel like instead of having null, we can have a particular string value stating it's background.

e.g `ELIGIBLE_NOT_RECEIVED`

Reply:
Do you mean having a const for `ELIGIBLE_NOT_RECEIVED` but still have it defined as `null` or have some other value associated with it?

Reply:
Added a constant, please take a look.

Reply:
Constant removed as per the conversation.